### DISCUSSION TITLE
can it auto scan the manuel key?

### DISCUSSION THREAD

**Discussion by ghost** [akopachov/flipper-zero_authenticator#75] [Q&A] [answered]
can it auto scan the manuel key?

edit: how to enter it (i dont understand)

edit2: if i do my discord account (the code is invaled)

**Comment by akopachov** [comment]
Hey, I don't quite understand what "manuel" key you are talking about. If you mean "token secret" then there are 3 ways how to add that token secret into the app.

Consider reading these wiki-pages:

- [What is token secret and where to get it?](https://github.com/akopachov/flipper-zero_authenticator/wiki/What-is-token-secret-and-where-to-get-it%3F)
- [How to add new token?](https://github.com/akopachov/flipper-zero_authenticator/wiki/How-to-add-new-token%3F)

**Comment by ghost** [comment, marked as answer]
okey what is sha1 - sha256 what to use

**Reply by akopachov** [reply]
SHA1, SHA256 and SHA512 are hashing algorithms. Which one to use depends on your service provider. In 99% - it is SHA1, which is default option in the app. So if you are not so sure which one to use - use SHA1 (default)

**Reply by ghost** [reply]
what if it gives invaled code

**Reply by akopachov** [reply]
Consider reading this wiki page [Flipper Authenticator generates invalid codes, why so?](https://github.com/akopachov/flipper-zero_authenticator/wiki/Flipper-Authenticator-generates-invalid-codes%2C-why-so%3F)

**Reply by ghost** [reply]
i am trying a change sha but it does not work and the Token secret is copy to the cli

**Reply by akopachov** [reply]
Have you checked if your Flipper device has precise clock? Have you checked timezone settings? Just like wiki says?

**Reply by ghost** [reply]
yes and that is fully correct 

sorry i am new to this

**Reply by akopachov** [reply]
What is service provider you are trying to setup 2FA for?

**Reply by ghost** [reply]
at this moment google and discord

**Reply by akopachov** [reply]
For both of them you need to use default values, SHA1 as a hashing algo,, 30sec token duration, 6-digits code.
So, just use `totp add <token_name>` without providing any other options and then make sure you are providing correct token secret.
After you added token - make sure your timezone offset is correctly set in the app.
After that ensure that clock on your flipper device is precise.

What is your timezone offset btw?

**Reply by ghost** [reply]
![image](https://user-images.githubusercontent.com/119863571/219054878-aad86859-1fab-4ab8-8209-bbdf0e3d344d.png)

**Reply by akopachov** [reply]
so you did not correctly set timezone as wiki says. Take a look at [this wiki section](https://github.com/akopachov/flipper-zero_authenticator/wiki/Flipper-Authenticator-generates-invalid-codes%2C-why-so%3F#timezone-is-not-correct)

**Reply by ghost** [reply]
![image](https://user-images.githubusercontent.com/119863571/219056186-0adcf5e0-393c-41b7-b49b-537cb53d54eb.png)

**Reply by akopachov** [reply]
According to `totp tz` command [definition](https://github.com/akopachov/flipper-zero_authenticator/wiki/Is-there-a-CLI%3F) you have to provide timezone offset in hours (not just timezone name).

**Reply by ghost** [reply]
thanks that was the problem

thanks for your time

**Reply by akopachov** [reply]
No problem. Glad it started working for you!
